gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -151,7 +151,7 @@ Licensed to the Apache Software Foundation (ASF) under one or more
             Pattern.compile(".*PUT\\s+INTO\\s+(TABLE)?.*SELECT\\s+.*",
                 Pattern.CASE_INSENSITIVE);
         private static final Pattern EXECUTION_ENGINE_STORE_HINT =
-            Pattern.compile(".*EXECUTIONENGINE(\\s+)?+=(\\s+)?+STORE\\s+.*",
+            Pattern.compile(".*EXECUTIONENGINE(\\s+)?+=(\\s+)?+STORE.*[\\r\\n].*",
```

Comment:
Not clear what you mean.

1) Which pattern are you talking about: the one in original code, your modification or the one I suggested

2) The pattern you have "[\\r\\n]" will match newlines/returns. To match a string "\n" I think you will need something like: ```(\\\\r|\\\\n)```